### PR TITLE
Implement and test PKCS#11 C_GetMechanismInfo

### PR DESCRIPTION
**Pull request by dcgaws**
This is an example of the additional test output log lines:
4 6729 [TestRunner] The PKCS #11 module supports RSA signing.
5 6729 [TestRunner] The PKCS #11 module supports ECDSA.
6 6729 [TestRunner] The PKCS #11 module supports elliptic-curve key generation.
TEST(Full_PKCS11_Capabilities, AFQP_Capabilities) PASS

### CODE REVIEW

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 1]
```diff
@@ -827,6 +827,43 @@ CK_DECLARE_FUNCTION( CK_RV, C_GetTokenInfo )( CK_SLOT_ID slotID,
...
+    CK_RV xResult = CKR_MECHANISM_INVALID;
+    struct CryptoMechanisms
+    {
+        CK_MECHANISM_TYPE xType;
+        CK_MECHANISM_INFO xInfo;
+    }
+    pxSupportedMechanisms[] =
+    {
+        { CKM_RSA_PKCS,        { 2048, 2048, 0 } },
+        { CKM_ECDSA,           { 256,  256,  0 } },
```
Comment (alexa-noxon) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c]: We also support CKM_SHA256, CKM_RSA_X_509

Comment (alexa-noxon) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c]: We should probably specify flags as well (like CKF_SIGN, CKF_DIGEST)

File: libraries/abstractions/pkcs11/test/iot_test_pkcs11.c [hunk 2]
```diff
@@ -700,7 +734,56 @@ TEST( Full_PKCS11_StartFinish, AFQP_OpenSessionCloseSession )
...
+    }
+
+    /* Check for elliptic-curve key generation support. */
+    xResult = pxGlobalFunctionList->C_GetMechanismInfo( pxSlotId[ 0 ], CKM_EC_KEY_PAIR_GEN, &MechanismInfo );
+    TEST_ASSERT_TRUE( CKR_OK == xResult || CKR_MECHANISM_INVALID == xResult );
+
+    if( CKR_OK == xResult )
+    {
+        configPRINTF( ( "The PKCS #11 module supports elliptic-curve key generation.\r\n" ) );
+    }
```
Comment (alexa-noxon) [inline libraries/abstractions/pkcs11/test/iot_test_pkcs11.c]: Should we also assert that P256 and 2048 rsa are within the key size info ranges?

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 3]
```diff
@@ -827,6 +827,43 @@ CK_DECLARE_FUNCTION( CK_RV, C_GetTokenInfo )( CK_SLOT_ID slotID,
     return CKR_OK;
 }
 
+/**
+ * @brief This function obtains information about a particular
+ * mechanism possibly supported by a token.
+ *
```
[the remark this answers is not in the document]
Comment (dcgaws) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c]: Yes, thank you, done.

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 4]
```diff
@@ -827,6 +827,53 @@ CK_DECLARE_FUNCTION( CK_RV, C_GetTokenInfo )( CK_SLOT_ID slotID,
...
+        { CKM_RSA_PKCS,        { 2048, 2048, CKF_SIGN              } },
+        { CKM_RSA_X_509,       { 2048, 2048, CKF_VERIFY            } },
+        { CKM_ECDSA,           { 256,  256,  CKF_SIGN | CKF_VERIFY } },
+        { CKM_EC_KEY_PAIR_GEN, { 256,  256,  CKF_GENERATE_KEY_PAIR } },
+        { CKM_SHA256,          { 0,    0,    CKF_DIGEST            } }
+    };
+    uint32_t ulMech = 0;
+
+    /* Look for the requested mechanism in the above table. */
+    for( ; ulMech < sizeof( pxSupportedMechanisms ) / sizeof( pxSupportedMechanisms[ 0 ] ); ulMech++ )
```
Comment (sarenameas) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c]: np lol
```
for( ; ulMech < ( sizeof( pxSupportedMechanisms ) / sizeof( pxSupportedMechanisms[ 0 ] ) ); ulMech++ )
```

**Pull request merged**